lib/taskana-core/src/main/java/pro/taskana/workbasket/internal/WorkbasketServiceImpl.java

```diff
@@ -176,6 +205,20 @@ public Workbasket updateWorkbasket(Workbasket workbasketToUpdate)
 
         workbasketMapper.update(workbasketImplToUpdate);
       }
+
+      if (HistoryEventManager.isHistoryEnabled()) {
+
+        String details =
+            ObjectAttributeChangeDetector.determineChangesInAttributes(
+                oldWorkbasket, workbasketToUpdate);
+
+        historyEventManager.createEvent(
+            new WorkbasketUpdatedEvent(
```

Comment:
Would it make sense to move the ID generation and the change detection into the history?

Reply:
ID generation maybe, but the change detection is very dependent on the actual method where it takes place because of different parameter objects getting passed to the change detector. While it's possible, this change would require several hours of rework.